classes/manager.php

```diff
@@ -628,6 +628,18 @@ public static function possible_factor_setup() {
         // Get all active factors.
         $factors = \tool_mfa\plugininfo\factor::get_enabled_factors();
 
+        // Check if there are enough factors that can ONLY pass, if so, don't display.
+        $weight = 0;
+        foreach ($factors as $factor) {
+            $states = $factor->possible_states($USER);
+            if (count($states) == 1 && reset($states) == \tool_mfa\plugininfo\factor::STATE_PASS) {
```

Comment:
I feel like this loop and the one below should be combined, the intent I think needs more commenting